src/Microsoft.AspNet.BasicMiddleware/HttpMethodOverrideMiddleware.cs

```diff
@@ -0,0 +1,40 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNet.Builder;
+using Microsoft.AspNet.Http;
+
+namespace Microsoft.AspNet.BasicMiddleware
+{
+    // You may need to install the Microsoft.AspNet.Http.Abstractions package into your project
+    public class HttpMethodOverrideMiddleware
+    {
+        private const string XHttpMethodOverride = "X-Http-Method-Override";
+        private readonly RequestDelegate _next;
+
+        public HttpMethodOverrideMiddleware(RequestDelegate next)
+        {
+            _next = next;
+        }
+
+        public Task Invoke(HttpContext context)
+        {
+            var xHttpMethodOverrideValue = context.Request.Headers.Get(XHttpMethodOverride);
+            if(xHttpMethodOverrideValue != null)
+            {
+                context.Request.Method = xHttpMethodOverrideValue;
+            }
+            return _next(context);
+        }
+    }
+
+    // Extension method used to add the middleware to the HTTP request pipeline.
+    public static class HttpMethodOverrideExtensions
```

Comment:
1. Don't add extension methods in a public nested class
2. Use Xml comments for documentation
3. The comment is wrong. It says "Extension method" but that's a class